flixel/graphics/FlxAsepriteUtil.hx

```diff
@@ -0,0 +1,194 @@
+package flixel.graphics;
+
+import flixel.graphics.frames.FlxAtlasFrames;
+import flixel.system.FlxAssets;
+
+class FlxAsepriteUtil
+{
+	public static function loadAseAtlas(sprite:FlxSprite, graphic, data:FlxAsepriteJsonAsset, tagSuffix:String = ":")
+	{
+		final aseData = data.getData();
+		sprite.frames = FlxAtlasFrames.fromAseprite(graphic, aseData);
+		
+		for (frameTag in aseData.meta.frameTags)
+			sprite.animation.addByPrefix(frameTag.name, frameTag.name + tagSuffix);
+	}
+}
+
+typedef AtlasBase<T> = { frames:T }
+
+typedef AseAtlasBase<T> = AtlasBase<T> &
+{
+	var meta:AseAtlasMeta;
+}
+
+typedef AseAtlasArray = AseAtlasBase<Array<AseAtlasFrame>>;
+typedef TexturePackerAtlasArray = AtlasBase<Array<AseAtlasFrame>>;
+
+abstract Hash<T>(Dynamic)
+{
+	public inline function keyValueIterator():KeyValueIterator<String, T>
+	{
+		var keys = Reflect.fields(this).iterator();
+		return
+		{
+			hasNext: keys.hasNext,
+			next: ()->
+			{
+				final key = keys.next();
+				return { key:key, value:Reflect.field(this, key)};
+			}
+		};
+	}
+}
+
+typedef AseAtlasHash = AseAtlasBase<Hash<AseAtlasFrame>>;
+typedef TexturePackerAtlasHash = AtlasBase<Hash<TexturePackerAtlasFrame>>;
+
+typedef HashOrArray<T> = flixel.util.typeLimit.OneOfTwo<Hash<T>, Array<T>>;
+
+typedef AseAtlas = AseAtlasBase<HashOrArray<AseAtlasFrame>>;
+typedef TexturePackerAtlas = AtlasBase<HashOrArray<TexturePackerAtlasFrame>>;
+
+typedef AseAtlasMeta =
+{
+	var app:String;
+	var version:String;
+	var image:String;
+	var format:String;
+	var size:AtlasSize;
+	var scale:String;
+	var frameTags:Array<AseAtlasTag>;
+	var slices:Array<AseAtlasSlice>;
+	var layers:Array<AseAtlasLayer>;
+}
+
+typedef AseAtlasTag = { name: String, from:Int, to:Int, direction:String };
+
+typedef AseAtlasSlice =
+{
+	var name:String;
+	var color:String;
+	
+	/**
+	 * Info of at what frames the slice changes size
+	 */
+	var keys: Array<AseAtlasSliceKey>;
+}
+
+typedef AseAtlasLayer =
+{
+	var name:String;
+	
+	/**
+	 * The name of the parent layer
+	 */
+	@:optional var group:String;
+	
+	/**
+	 * Ranges from 0 to 255
+	 */
+	@:optional var opacity:Int;
+	
+	/**
+	 * The effect added to this layer to change how the colors blend with colors on lower layers
+	 * @see AseBlendMode
+	 */
+	@:optional var blendMode:AseBlendMode;
+	
+	/**
+	 * Any notes that were left on cels of this layer
+	 */
+	@:optional var cels:Array<{ frame:Int, data:String, ?color:String }>;
```

Comment:
I failed to update my Aseprite to the latest 1.2x and am now on the release candidate for 1.3. It looks like they've added a z-index to the cel data. Also per your update to make `color` optional, it appears that `data` is also not included if it is empty in aseprite.

So I believe this is more accurate:

```
/**
 * Any properties on cels of this layer
 */
@:optional var cels:Array<{ frame:Int, ?zIndex:Int, ?data:String, ?color:String }>;
```

This is pulled from the export json with 3 frames, each having only one of the 3 values set:

```
"cels": [
    { "frame": 1, "zIndex": 4 },
    { "frame": 2, "color": "#8f8db7d8" },
    { "frame": 3, "data": "user data" }
]
```

https://www.aseprite.org/api/cel#celzindex

Reply:
Let me grab latest aseprite, thanks for all the feedback, so far!